src/snippets/responsive-image.liquid

```diff
@@ -0,0 +1,67 @@
+{% comment %}
+    It creates a style tag and it restricts an image from growing larger than its max resolution.
+
+    Dependencies:
+    - Lazysizes plugin (https://github.com/aFarkas/lazysizes) which enable responsive image with faster load time and better performance.
+    - Lazysizes Responsive Images as a Service (https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/rias) To load the correct image size with our CDN
+    - Lazysizes Bgset (https://github.com/aFarkas/lazysizes/tree/gh-pages/plugins/bgset) To use responsive images on background-image (CSS)
+
+    Accepts:
+    - image: {Object} Image object
+    - width: {Number} Max width of the image container
+    - height: {Number} Max height of the image container
+    - img_class: {String} class name of the <img />
+    - wrapper_class: {String} class name of the <div> wrapper
+
+    Usage:
+    In your liquid template file, copy the following line
+    - {% include 'responsive-image' with image: featured_image, img_class: "css-class", wrapper_class: "wrapper-css-class", width: 700, height: 800 %}
+{% endcomment %}
+
+{% comment %} Added incremental number to avoid conflict styling code when the same images are using this snippet {% endcomment %}
+{% capture responsive_image_counter %}{% increment image_counter %}{% endcapture %}
+
+<style>
+  {% comment %} If the image is portrait {% endcomment %}
+  {% if image.aspect_ratio <= 1 %}
+    {% assign max_height = image.height | at_most: height %}
+    {% assign max_width = max_height | times: image.aspect_ratio %}
+  {% else %}
+    {% assign max_width = image.width | at_most: width %}
+    {% assign max_height = max_width | divided_by: image.aspect_ratio %}
+  {% endif %}
+
+  #ProductImage-{{ image.id }}-{{ responsive_image_counter }} {
+    max-width: {{ max_width }}px;
+    max-height: {{ max_height }}px;
+  }
+  #ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }} {
+    max-width: {{ max_width }}px;
+  }
+
+  #ProductImageWrapper-{{ image.id }}-{{ responsive_image_counter }}:before {
+    {% comment %}
```

Comment:
Small nit: Mind switch this to `{%- comment -%}` to reduce the big chunks of whitespace in the HTML? 

[Whitespace control in Liquid](https://help.shopify.com/themes/liquid/basics/whitespace)

Reply:
Thank you for the tips!